### PR TITLE
[DT-4] Added sniffs for generating "get..." methods for the core

### PR DESCRIPTION
**Pull request by pushokwhite**
#### Overview

- Ticket: https://spryker.atlassian.net/browse/DT-4

###### Change log

Added sniffs for generating "get..." methods for Facades, Factories, Types, CollectionTypes, Controllers, Consoles etc.

### CODE REVIEW

File: Spryker/Sniffs/CollectionType/FacadeMethodAnnotationSniff.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * @inheritdoc
+     */
+    public function process(File $phpCsFile, $stackPointer)
+    {
+        if (!$this->isCollectionType($phpCsFile, $stackPointer)) {
+            return;
+        }
+
+        $bundle = $this->getModule($phpCsFile);
```
Comment (dereuromark) [inline Spryker/Sniffs/CollectionType/FacadeMethodAnnotationSniff.php]: $module

File: Spryker/Sniffs/CollectionType/FactoryMethodAnnotationSniff.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    /**
+     * @inheritdoc
+     */
+    public function process(File $phpCsFile, $stackPointer)
+    {
+        if (!$this->isCollectionType($phpCsFile, $stackPointer)) {
+            return;
+        }
+
+        $bundle = $this->getModule($phpCsFile);
```
Comment (dereuromark) [inline Spryker/Sniffs/CollectionType/FactoryMethodAnnotationSniff.php]: same and others

**Comment by dereuromark** [comment]
Those are about doc blocks, right?
As such maybe putting them from

    Spryker/Sniffs/CollectionType/ 

into

    Spryker/Sniffs/Commenting/ 

?

**Comment by pushokwhite** [comment]
Sure but for this moment as we discussed we are not refactoring the code, only making it workable. As we already got this format like the Class type identifying what sniffs should it pass - I've decided to create everything regarding this style.

**Comment by dereuromark** [comment]
after fixing small comments.

**Comment by dereuromark** [comment]
Please also run `docs` command to generate updated docs.md file.

File: docs/README.md [hunk 3]
```diff
@@ -79,12 +79,17 @@ SlevomatCodingStandard (20 sniffs)
 - SlevomatCodingStandard.TypeHints.ReturnTypeHintSpacing
 - SlevomatCodingStandard.Variables.DuplicateAssignmentToVariable
 
-Spryker (67 sniffs)
+Spryker (94 sniffs)
```
Comment (dereuromark) [inline docs/README.md]: Did we really add 27 sniffs? :open_mouth:

Comment (pushokwhite) [reply docs/README.md]: Yeah, me too, but for now we have no annotation-lead sniffers, we got one sniffer per annotation + class (Factory, Config, Controller) combination.

Comment (dereuromark) [reply docs/README.md]: Thats not ideal. We also have to think about the implications to runtime here for so many more sniffs, if we go over all 500+ modules all the time.

**Comment by pushokwhite** [comment]
Already updated in the second commit.

File: Spryker/Sniffs/AbstractSniffs/AbstractClassDetectionSprykerSniff.php [hunk 4]
```diff
@@ -77,139 +38,87 @@ protected function extendsAbstractController(File $phpCsFile, int $stackPointer)
 
     /**
      * @param \PHP_CodeSniffer\Files\File $phpCsFile
-     * @param int $stackPointer
+     * @param string $predefinedName
      *
      * @return bool
      */
-    protected function isCollectionType(File $phpCsFile, int $stackPointer): bool
+    protected function hasCorrectName(File $phpCsFile, string $predefinedName)
```
Comment (dereuromark) [inline Spryker/Sniffs/AbstractSniffs/AbstractClassDetectionSprykerSniff.php]: where did the :bool typehint go?

Comment (pushokwhite) [reply Spryker/Sniffs/AbstractSniffs/AbstractClassDetectionSprykerSniff.php]: My bad, returning..

**Pull request merged**